src/packages/block/block-list/components/block-list-entry/block-list-entry.element.ts

```diff
@@ -252,40 +261,53 @@ export class UmbBlockListEntryElement extends UmbLitElement implements UmbProper
 				>${this._inlineEditingMode ? this.#renderInlineBlock() : this.#renderRefBlock()}</umb-extension-slot
 			>
 			<uui-action-bar>
-				${this._showContentEdit && this._workspaceEditContentPath
-					? html`<uui-button
-							label="edit"
-							look="secondary"
-							color=${this._contentInvalid ? 'danger' : ''}
-							href=${this._workspaceEditContentPath}>
-							<uui-icon name="icon-edit"></uui-icon>
-							${this._contentInvalid
-								? html`<uui-badge attention color="danger" label="Invalid content">!</uui-badge>`
-								: nothing}
-						</uui-button>`
-					: nothing}
-				${this._hasSettings && this._workspaceEditSettingsPath
-					? html`<uui-button
-							label="Edit settings"
-							look="secondary"
-							color=${this._settingsInvalid ? 'danger' : ''}
-							href=${this._workspaceEditSettingsPath}>
-							<uui-icon name="icon-settings"></uui-icon>
-							${this._settingsInvalid
-								? html`<uui-badge attention color="danger" label="Invalid settings">!</uui-badge>`
-								: nothing}
-						</uui-button>`
-					: nothing}
-				<uui-button label="delete" look="secondary" @click=${() => this.#context.requestDelete()}>
-					<uui-icon name="icon-remove"></uui-icon>
-				</uui-button>
+				${this.#renderEditContentAction()} ${this.#renderEditSettingsAction()} ${this.#renderDeleteAction()}
 			</uui-action-bar>
 			${!this._showContentEdit && this._contentInvalid
 				? html`<uui-badge attention color="danger" label="Invalid content">!</uui-badge>`
 				: nothing}
 		`;
 	}
 
+	#renderEditContentAction() {
+		return html` ${this._showContentEdit && this._workspaceEditContentPath
+			? html`<uui-button
+					label="edit"
+					look="secondary"
+					color=${this._contentInvalid ? 'danger' : ''}
+					href=${this._workspaceEditContentPath}>
+					<uui-icon name="icon-edit"></uui-icon>
+					${this._contentInvalid
+						? html`<uui-badge attention color="danger" label="Invalid content">!</uui-badge>`
+						: nothing}
+				</uui-button>`
+			: nothing}`;
+	}
+
+	#renderEditSettingsAction() {
+		return html`
+			${this._hasSettings && this._workspaceEditSettingsPath
+				? html`<uui-button
+						label="Edit settings"
+						look="secondary"
+						color=${this._settingsInvalid ? 'danger' : ''}
+						href=${this._workspaceEditSettingsPath}>
+						<uui-icon name="icon-settings"></uui-icon>
+						${this._settingsInvalid
+							? html`<uui-badge attention color="danger" label="Invalid settings">!</uui-badge>`
+							: nothing}
+					</uui-button>`
+				: nothing}
+		`;
+	}
+
+	#renderDeleteAction() {
+		if (this.readonly) return nothing;
```

Comment:
This seems not to work, unless I've done something wrong.

Its still rendered, despite begin in readonly mode.
(the way i got to it was by tabbing into the element. But I just wanted to highlight that this line does not prevent the rendering of this action)

I would also expect to be able to navigate into Blocks. With the overlay i get currently I cant. It also does not seem like the Block Workspace got the readonly state.

![image](https://github.com/user-attachments/assets/2a083f4f-71f6-4477-9330-11697f483f96)

Reply:
Here a screenshot from tabbing to open the Block Workspace. Fields did not get readonly.
![image](https://github.com/user-attachments/assets/bc95f996-501f-44aa-b605-da734a51b3e3)

Reply:
it works now